src/components/Amplitude.js

```diff
@@ -1,149 +1,43 @@
-import debounce from 'lodash.debounce';
-import PropTypes from 'prop-types';
 import React from 'react';
-import shallowequal from 'shallowequal';
-
-import { memoize } from '../lib/memoize';
-import { isValidAmplitudeInstance } from '../lib/validation';
-
-class Amplitude extends React.Component {
-  constructor(props) {
-    super(props);
-
-    if (typeof props.debounceInterval === 'number') {
-      this.logEvent = debounce(this._makeLogEvent(), props.debounceInterval);
-    } else {
-      this.logEvent = this._makeLogEvent();
-    }
-
-    this._renderPropParams = {
-      logEvent: this.logEvent,
-      instrument: this.instrument,
-    };
-  }
-
-  _makeLogEvent = () => (eventType, eventProperties, callback) => {
-    const amplitudeInstance = this.getAmplitudeInstance();
-
-    if (amplitudeInstance) {
-      amplitudeInstance.logEvent(
-        eventType,
-        {
-          ...this.getAmplitudeEventProperties(),
-          ...(eventProperties || {}),
-        },
-        callback,
-      );
-    }
-  };
-
-  instrument = memoize((eventType, func) => {
-    return (...params) => {
-      const retVal = func ? func(...params) : undefined;
-
-      this.logEvent(eventType);
-
-      return retVal;
-    };
-  });
-
-  componentWillReceiveProps(nextProps) {
-    const { props } = this;
-
-    if (typeof nextProps.debounceInterval === 'number') {
-      if (props.debounceInterval !== nextProps.debounceInterval) {
-        this.logEvent = debounce(this._makeLogEvent(), nextProps.debounceInterval);
-
-        this._renderPropParams = {
-          ...this._renderPropParams,
-          logEvent: this.logEvent,
-        };
-      }
-    } else if (typeof props.debounceInterval === 'number') {
-      this.logEvent = this._makeLogEvent();
-
-      this._renderPropParams = {
-        ...this._renderPropParams,
-        logEvent: this.logEvent,
-      };
-    }
-  }
-
-  componentDidMount() {
-    const { props } = this;
-    const amplitudeInstance = this.getAmplitudeInstance();
-
-    if (amplitudeInstance && props.userProperties) {
-      amplitudeInstance.setUserProperties(props.userProperties);
-    }
-  }
-
-  componentDidUpdate(prevProps) {
-    this.instrument.cache.garbageCollect();
-
-    const { props } = this;
-    const amplitudeInstance = this.getAmplitudeInstance();
-
-    if (amplitudeInstance && !shallowequal(prevProps.userProperties, props.userProperties)) {
-      amplitudeInstance.setUserProperties(props.userProperties);
-    }
-  }
+import PropTypes from 'prop-types';
 
-  getChildContext() {
+import { AmplitudeContext } from './AmplitudeProvider';
+import { useInstrument } from '../hooks/use-instrument';
+import { useLogEvent } from '../hooks/use-log-event';
+import { useSetUserProperties } from '../hooks/use-set-user-properties';
+
+const Amplitude = ({
```

Comment:
This component act as render prop amplitude provider - this still seems useful, even when havings hooks and all. You could wrap this logic in `useAmplitude` hook to provide this functionality for hook users and just reuse this here to provide render prop API on top of that.